src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -394,7 +396,7 @@ private ProjectInfo CreateProjectInfoForDocument(string filePath)
             var documentInfo = DocumentInfo.Create(
                 documentId,
                 filePath,
-                sourceCodeKind: parseOptionsOpt?.Kind ?? SourceCodeKind.Regular,
+                sourceCodeKind: parseOptionsOpt?.Kind ?? (string.Equals(fileExtension, languageInformation.ScriptExtension, StringComparison.OrdinalIgnoreCase) ? SourceCodeKind.Script : SourceCodeKind.Regular),
```

Comment:
make sure we let F# or any other language to set document kind as script if they want through "ScriptExtension" they already provide to us. previously, we only let us to set document script.

Reply:
Can we just extract the SourceCodeKind out to a variable and set that in the "if script" block above? Maybe restructure this as

```
if (fileExtension == languageInformation.ScriptExtension)
{
    kind = SourceCodeKind.Script;
    if (parseOptions != null && compilationOptions != null)
    {
        // do the C#/VB specific stuff
    }
}

Reply:
👍   This code is too complex.

Reply:
sure.